src/test/java/core/basesyntax/RegistrationServiceTest.java

```diff
@@ -0,0 +1,138 @@
+package core.basesyntax;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.InvalidDataException;
+import core.basesyntax.model.User;
+import core.basesyntax.service.RegistrationService;
+import core.basesyntax.service.RegistrationServiceImpl;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static User bob;
+    private static User alice;
+    private static User max;
+    private RegistrationService registrationService = new RegistrationServiceImpl();
+
+    @BeforeAll
+    static void beforeAll() {
+        bob = new User();
+        bob.setLogin("bob");
+        bob.setPassword("1234567");
+        bob.setAge(21);
+        alice = new User();
+        alice.setLogin("alice");
+        alice.setPassword("456789");
+        alice.setAge(27);
+        max = new User();
+    }
+
+    @AfterEach
+    void tearDown() {
+        Storage.people.clear();
+    }
+
+    @Test
+    void registration_userWithValidData_Ok() {
+        assertEquals(bob, registrationService.register(bob));
+    }
+
+    @Test
+    void registration_twoUsersWithValidData_Ok() {
+        registrationService.register(bob);
+        registrationService.register(alice);
+        int actual = Storage.people.size();
+        assertEquals(2, actual);
+    }
+
+    @Test
+    void registration_userWithSameLogin_notOk() {
+        registrationService.register(bob);
+        registrationService.register(alice);
+        max.setLogin("alice");
+        max.setPassword("789000");
+        max.setAge(20);
+        assertThrows(InvalidDataException.class, () -> registrationService.register(max));
+    }
+
+    @Test
+    void registration_userAgeUnder18_notOk() {
+        max.setLogin("max");
+        max.setPassword("789123");
+        max.setAge(17);
```

Comment:
you can create it in the method beforeAll too and just change some values that you need to test